ngraph/core/src/pass/convert_precision.cpp

```diff
@@ -0,0 +1,830 @@
+                             ngraph::element::Type to,
+                             size_t idx)
+{
+    if (auto type_relaxed = std::dynamic_pointer_cast<op::TypeRelaxedBase>(node))
+    {
+        type_relaxed->set_overridden_output_type(to);
+        return true;
+    }
+    else if (auto casted = std::dynamic_pointer_cast<opset1::ShapeOf>(node))
+    {
+        auto relaxed_op = std::make_shared<ngraph::op::TypeRelaxed<opset1::ShapeOf>>(
+            *casted, element::TypeVector{}, element::TypeVector{to});
+        replace_node(node, relaxed_op);
+        return true;
+    }
+    return false;
+}
+
+bool extend_select_type(const std::shared_ptr<ngraph::Node>& node,
+                        ngraph::element::Type to,
+                        size_t idx)
+{
+    if (auto type_relaxed = std::dynamic_pointer_cast<op::TypeRelaxedBase>(node))
+    {
+        type_relaxed->set_origin_input_type(element::boolean, 0);
+        return true;
+    }
+    else if (auto casted = std::dynamic_pointer_cast<opset4::Select>(node))
+    {
+        auto relaxed_op = std::make_shared<op::TypeRelaxed<opset4::Select>>(
+            *casted, element::TypeVector{element::boolean}, element::TypeVector{});
+        replace_node(node, relaxed_op);
+        return true;
+    }
+    return false;
+}
+
+template <typename src_type, typename dst_type>
+inline dst_type convert_value(src_type val)
+{
+    if (val > std::numeric_limits<dst_type>::max())
+    {
+        return std::numeric_limits<dst_type>::max();
+    }
+    else if (val < std::numeric_limits<dst_type>::lowest())
+    {
+        return std::numeric_limits<dst_type>::lowest();
+    }
+    return static_cast<dst_type>(val);
+}
+
+// We need to treat U64->I32 and U32->I32 as a separate case, because of C++'s implicit promotion
+// from signed to unsigned, and we don't need to compare and clamp the input to
+// std::numeric_limits<int32_t>::lowest()
+template <>
+inline int32_t convert_value<uint64_t, int32_t>(uint64_t val)
+{
+    if (val > static_cast<uint64_t>(std::numeric_limits<int32_t>::max()))
+    {
+        return std::numeric_limits<int32_t>::max();
+    }
+    return static_cast<int32_t>(val);
+}
+
+template <>
+inline int32_t convert_value<uint32_t, int32_t>(uint32_t val)
+{
+    if (val > static_cast<uint32_t>(std::numeric_limits<int32_t>::max()))
+    {
+        return std::numeric_limits<int32_t>::max();
+    }
+    return static_cast<int32_t>(val);
+}
+
+namespace
+{
+    template <element::Type_t PREC_FROM, element::Type_t PREC_TO>
+    std::shared_ptr<ngraph::Node>
+        change_constant_precision(std::shared_ptr<opset4::Constant>& constant)
+    {
+        using src_type = typename element_type_traits<PREC_FROM>::value_type;
+        using dst_type = typename element_type_traits<PREC_TO>::value_type;
+
+        const auto* src_data = constant->get_data_ptr<src_type>();
+        const auto size = shape_size(constant->get_shape());
+
+        auto new_constant =
```

Comment:
should we copy runtime info to new const?

Reply:
We do not copy rt info for Constant operations.